crates/typst-cli/src/world.rs

```diff
@@ -76,11 +81,26 @@ impl SystemWorld {
         let main_path = VirtualPath::within_root(&input, &root)
             .ok_or("input file must be contained in project root")?;
 
+        let mut inputs = HashMap::with_capacity(command.plain_inputs.len());
+        for (k, v) in &command.plain_inputs {
+            inputs.insert(k.clone().into(), Value::Str(v.clone().into()));
+        }
+        for raw in &command.json_inputs {
+            let root: HashMap<EcoString, Value> =
+                serde_json::from_str(raw).map_err(|e| eco_format!("{e}"))?;
+            inputs.reserve(root.len());
+            for (key, val) in root {
+                inputs.insert(key, val);
+            }
+        }
```

Comment:
This could be done more idiomatically with iterators.

Reply:
Iterators get kind of annoying here because from_iter requires annotations to deduce the RandomState type, but I'll try to see if I can find a nicer way

Reply:
That's strange. I'm surprised you can't just do `.collect::<HashMap<EcoString, Value>>()`

Reply:
> That's strange. I'm surprised you can't just do `.collect::<HashMap<EcoString, Value>>()`

I believe it should be possible (without using `Dict::from_iter`):

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7007968ee2585f2a7e800087bd5e9d7f

`.collect()` is a _very_ powerful beast in Rust (see also [a cool fun fact about .collect()](https://ntietz.com/blog/rust-vec-of-result/)).

Reply:
That collecting iterator of Results to Result of a container thing will be necessary here considering the possibility of failure for the JSON deserialization.